File: src/Web/Dtos/MemberModuleDto.cs
namespace Web.Dtos;

public class MemberModuleDto
{
    public string ModuleId { get; set; } = null!;
    public string? Name { get; set; }
    public string? Subject { get; set; }
    public string? CardImageUrl { get; set; }
    public int ProgressPercent { get; set; }
    public bool IsCompleted { get; set; }
}

File: src/Web/vue-app/src/types/entities/memberModuleDto.ts
export interface MemberModuleDto {
  moduleId: string;
  // Backend currently exposes non-localized fields.
  name?: string;
  subject?: string;
  nameFr?: string;
  nameEn?: string;
  name?: string;
  sujetFr?: string;
  sujetEn?: string;
  subject?: string;
  cardImageUrl?: string;
  assignedAt?: string;
  progressPercent: number;
  isCompleted: boolean;
}
